sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -608,8 +608,11 @@ AllocaCommandBase *Scheduler::GraphBuilder::getOrCreateAllocaForReq(
       AllocaCmd = new AllocaSubBufCommand(Queue, *Req, ParentAlloca);
       UpdateLeafs(findDepsForReq(Record, Req, Queue), Record,
                   access::mode::read_write);
-    } else
-      AllocaCmd = new AllocaCommand(Queue, FullReq);
+    } else {
+      // Can reuse user data for the first allocation
+      bool InitFromUserData = Record->MAllocaCommands.empty();
+      AllocaCmd = new AllocaCommand(Queue, FullReq, InitFromUserData);
```

Comment:
Minor: since we can only reuse user pointer on the first allocation, I suggest also either getting rid of default value for the third argument of AllocaCommand constructor or changing it to false.

Reply:
Done.